com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -1099,12 +1180,34 @@ public virtual void CollectDiscreteActionMasks(DiscreteActionMasker actionMasker
         ///
         /// [Agents - Actions]: https://github.com/Unity-Technologies/ml-agents/blob/release_5_docs/docs/Learning-Environment-Design-Agents.md#actions
         /// </remarks>
-        /// <param name="vectorAction">
-        /// An array containing the action vector. The length of the array is specified
-        /// by the <see cref="BrainParameters"/> of the agent's associated
-        /// <see cref="BehaviorParameters"/> component.
+        /// <param name="actions">
+        /// Struct containing the buffers of actions to be executed at this step.
         /// </param>
-        public virtual void OnActionReceived(float[] vectorAction) {}
+        public virtual void OnActionReceived(ActionBuffers actions)
+        {
+            // Copy the actions into our local array and call the original method for
+            // backward compatibility.
+            // For now we need to check which array has the actions in them in order to pass it back to the old method.
+            if (actions.ContinuousActions.Length > 0)
```

Comment:
Would it be cleaner and/or easier to test if we made this a utility method of ActionBuffers?

Reply:
i wanted to keep it here to keep the context of "this is here strictly for backward compatibility", and I'm willing to move it if you feel strongly about it.